--- include/gsl/gsl_util
+++ include/gsl/gsl_util
@@ -61,5 +61,11 @@
     final_action(const final_action&) = delete;
     final_action& operator=(const final_action&) = delete;
-    final_action& operator=(final_action&&) = delete;
+    final_action& operator=(final_action&& other) noexcept
+    {
+        f_ = std::move(other.f_);
+        invoke_ = other.invoke_;
+        other.invoke_ = false;
+        return *this;
+    }

     GSL_SUPPRESS(f.6) // NO-FORMAT: attribute // terminate if throws
